File: 20230802/java/hyeonjeong.java
import java.io.IOException;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class Solution {

    static int building, road;
    static ArrayList<Info> map[];
    static int distance[];
    static Queue<Info> queue;
    
    static class Info implements Comparable<Info>{
        int end, dist;
        public Info(int end, int dist) {
            this.end = end;
            this.dist = dist;
        }
        @Override
        public int compareTo(Info o) {
            return Integer.compare(this.dist, o.dist);
        }
    }
    public static void main(String[] args) throws NumberFormatException, IOException {
        Scanner sc = new Scanner(System.in);
        int tc = sc.nextInt();
        for(int t=1; t<=tc; t++) {
            building = sc.nextInt();
            road = sc.nextInt();

            map = new ArrayList[building+1];
            for(int i=1; i<=building; i++) {
                int start = sc.nextInt();
                int end = sc.nextInt();
                int dist = sc.nextInt();
                map[start].add(new Info(end, dist));
            }

            int answer = Integer.MAX_VALUE;
            for(int i=1; i<=building; i++) {
                distance = new int[building+1];
                queue = new LinkedList<Info>();
                for(int j=1; j<=building; j++) {
                    distance[j] = Integer.MAX_VALUE;
                }
                queue.add(new Info(i, 0));
                Info info;
                while(!queue.isEmpty()) {
                    info = queue.poll();
                    int cend = info.end;
                    int cdist = info.dist;
                    if(cdist > answer) continue;
                    for(int j=0; j<map[cdist].size(); j++) {
                        int nend = map[cend].get(j).end;
                        int ndist = map[cend].get(j).dist;
                        if(distance[nend] > cdist + ndist) {
                            distance[nend] = cdist + ndist;
                            queue.add(new Info(nend, distance[nend]));
                        }
                    }
                }
                answer = Math.min(distance[i], answer);
            }
            if(answer==Integer.MAX_VALUE) answer = -1;
            System.out.printf("#%d %d%n", t, answer);
        }
        
    }
}

File: 20230911/python/hyeonjeong.py
def solution(temperature, t1, t2, a, b, onboard):
    answer = 0
    t = temperature
    
    info = [[] for _ in range(t + len(onboard)+10)]
    for i in range(t-len(onboard)+1, t+len(onboard)):
        if i > temperature:
            dp[i].append([i-1, 0])
        elif i >= t1:
            dp[i].append([i-1, a])
        if i < temperature:
            dp[i].append([i+1, 0])
        elif i <= t2:
            dp[i].append([i+1, a])
        if i == temperature:
            dp[i].append([i, 0])
        elif t1 <= i <= t2:
            dp[i].append([i, b])
            
    dp = [[] for _ in range(len(onboard))]
    dp[0] = t
    for i in range(1, len(onboard)):
        for ct in dp[i-1]:
            for nt in info[i]:
                if onboard[i] and t1 <= nt <= t2:
                    dp.append(nt)
                elif not onboard[i]:
                    dp.append(nt)
          
    return answer
